util/loaders/TrainingLoader.ts

```diff
@@ -0,0 +1,227 @@
+      if (name !== null) {
+        const text = this.getTrainingRowValue(rowCounter, "B");
+        const item: any = {
+          name,
+          text,
+          type,
+          levels: [
+            //first level
+            {
+              level: level,
+              modX: this.getTrainingRowValue(rowCounter, "E"),
+              modY: this.getTrainingRowValue(rowCounter, "F"),
+              modZ: this.getTrainingRowValue(rowCounter, "G"),
+              gold: this.getTrainingRowValue(rowCounter, ids.gold),
+              books: this.getBooks(type, rowCounter, ids),
+              materials: this.getMaterials(rowCounter),
+            },
+          ],
+        };
+        resultArray.push(item);
+        arrayCounter++;
+      } else {
+        // Parse rest data and it into levels property
+        resultArray[arrayCounter].levels.push({
+          level: level,
+          modX: this.getTrainingRowValue(rowCounter, "E"),
+          modY: this.getTrainingRowValue(rowCounter, "F"),
+          modZ: this.getTrainingRowValue(rowCounter, "G"),
+          gold: this.getTrainingRowValue(rowCounter, ids.gold),
+          books: this.getBooks(type, rowCounter, ids),
+          materials: this.getMaterials(rowCounter),
+        });
+      }
+      rowCounter++;
+
+      notDone = !this.isLastRow(rowCounter, ids.name, ids.gold);
+    }
+
+    return this.generateMap(resultArray);
+  }
+
+  private generateMap(TrainingSKillArray: TrainingSKill[]): TrainingSKillMap {
+    const trainingSKillMap: TrainingSKillMap = {};
+
+    for (const skill of TrainingSKillArray) {
+      trainingSKillMap[skill.name] = skill;
+    }
+    return trainingSKillMap;
+  }
+
+  private getMaterials(rowCounter: number) {
+    const ids = this.mapColumnHeadersToColumnIds(
+      ANIKI_DROP_HEADERS,
+      this.workBook.Sheets.Training
+    ) as ANIKI_DROP_HEADERS_IDS;
+
+    const keys = Object.keys(ANIKI_DROP_HEADERS) as Array<
+      keyof typeof ANIKI_DROP_HEADERS
+    >;
+
+    const result: TrainingMaterial[] = [];
+    for (const key of keys) {
+      const count = this.getTrainingRowValue(rowCounter, ids[key]);
+
+      if (count) {
+        result.push({
+          name: ANIKI_DROP_HEADERS[key][0],
+          count: count,
+        });
+      }
+    }
+    return result;
+  }
+
+  private getBooks(
+    type: string,
+    rowCounter: number,
+    ids: TRAINING_HEADERS_IDS
+  ): TrainingMaterial[] {
+    const result: TrainingMaterial[] = [];
+    let prettyType = "";
+    switch (type) {
+      case "Infantry":
+        prettyType = "infantry";
+        break;
+      case "Lancer":
+        prettyType = "lancer";
+        break;
+      case "Cavalry":
+        prettyType = "cavalry";
+        break;
+      case "Flier/Aquatic":
+        prettyType = "flier";
+        break;
+      case "Archer/Assassin":
+        prettyType = "archer";
+        break;
+      case "Holy/Mage/Demon":
+        prettyType = "holy";
+        break;
+    }
```

Comment:
Magic string matching into switch statement, I would recommend making a type for these new strings and then making a mapping object and just plucking the value off it

```
const prettyType = prettyTypeMap[type]
```